shared/menubar/out-of-date.js

```diff
@@ -21,15 +21,24 @@ const OutOfDate = ({outOfDate, updateNow}: Props) =>
       centerChildren={true}
       direction="vertical"
     >
-      <Kb.Text type="BodySemibold" style={outOfDate.critical ? styles.textCritical : styles.textNonCritical}>
+      <Kb.Text
+        backgroundMode="Information"
+        type="BodySemibold"
+        style={outOfDate.critical ? styles.textCritical : undefined}
```

Comment:
I tried that initially but flow was complaining that style isn't compatible with a boolean.

Reply:
Really? I just tested locally and it worked okay (though, on a different component).

Reply:
Hrm; let me try again; maybe I mis-read the flow error.

Reply:
@songgao @adamjspooner Banners are using BodySmallSemibold now

Reply:
OK will update!

Reply:
@adamjspooner Yeah; still not working:

```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ menubar/out-of-date.js:41:18
Cannot create Kb.Text element because in property style:
 • Either inexact boolean [1] is incompatible with exact _StylesCrossPlatform [2].
 • Or boolean [1] is incompatible with read-only array type [3].
     menubar/out-of-date.js
     34│         style={outOfDate.critical && styles.textCritical}
     35│       >
     36│         Please{' '}
     37│         <Kb.Text
     38│           backgroundMode="Information"
     39│           type="BodySemibold"
     40│           underline={!!updateNow}
     41│           style={outOfDate.critical && styles.textCritical}
     42│           onClick={updateNow}
     43│         >
     44│           update now
     45│         </Kb.Text>
     46│         .
     47│       </Kb.Text>
     48│     </Kb.Box2>
     constants/types/config.js
 [1] 10│   critical: boolean,
     styles/css.js.flow
 [2] 96│   | _StylesCrossPlatform
 [3] 97│   | $ReadOnlyArray<_StylesCrossPlatform | null | false | void>
```

Reply:
Weird. I don't fully understand what an `inexact boolean` is. I thought that related to objects: https://devhints.io/flow#objects

Reply:
I think it's just that the only way `style` can take a boolean is when it's part of an array. i.e. it's OK to have `style={[false]}`, but not `style={false}`.

Reply:
I use this pattern (`style={props.someBoolean && styles.appliedStylesIfTrue`) all over the place, though. @chrisnojima, can you explain what's going on here?

Reply:
That works only in the contexts of collapseStyles (we explicitly allow Boolean and filter it out)

Reply:
Thanks! I'm sorry for the confusion, @songgao!

Reply:
🤔 So, which is the preferred method?

```
style={Styles.collapseStyles([someBoolean && someStyles])}
```
or:
```
style={someBoolean ? someStyles : undefined}
```

Reply:
Either is fine. CollapseStyles is to mix multiple styles so it's a little weird to use it in this case

Reply:
NP! Thanks for the discussion and clarification.